src/nix/profile.cc

```diff
@@ -232,54 +253,26 @@ struct CmdProfileInstall : InstallablesCommand, MixDefaultProfile
     {
         ProfileManifest manifest(*getEvalState(), *profile);
 
-        std::vector<DerivedPath> pathsToBuild;
+        auto builtPaths = Installable::build(getEvalStore(), store, Realise::Outputs, installables, bmNormal);
 
         for (auto & installable : installables) {
+            ProfileElement element;
+
             if (auto installable2 = std::dynamic_pointer_cast<InstallableFlake>(installable)) {
+                // FIXME: make build() return this?
```

Comment:
I do think what buildPaths returns could help with this or `updateStorePaths`? Each `BuiltPath` contains the paths for the outputs requested, so you don't need to do another `queryDerivationOutputMap` if you can use it.

Reply:
No, the issue here is that we need the resolved flakeref / attrpath.